feder/institutions/viewsets.py

```diff
@@ -19,15 +23,61 @@ class Meta:
         fields = ["name", "tags", "jst", "regon"]
 
 
+class InstitutionCSVRenderer(PaginatedCSVRenderer):
+    header = [
+        "pk",
+        "name",
+        "email",
+        "jst",
+        "jst_name",
+        "jst_voivodeship",
+        "jst_category",
+        "created",
+        "modified",
+        "regon",
+        "self",
+        "slug",
+        "tags.0",
+        "url",
+    ]
+    labels = {
+        "pk": _("Primary key"),
+        "name": _("Name"),
+        "email": _("Email of institution"),
+        "jst": _("JST id"),
+        "jst_name": _("JST name"),
+        "jst_category": _("JST category"),
+        "jst_voivodeship": _("JST voivodeship"),
+        "created": _("Created"),
+        "modified": _("Modified"),
+        "regon": _("REGON number"),
+        "self": _("Parent institutions"),
+        "slug": _("Slug"),
+        "tags.0": _("Tag"),
+        "url": _("URL"),
+    }
+
+
+class InstitutionPaginator(PageNumberPagination):
+    max_page_size = (
+        10000  # increased maximum page size to allow export to CSV without pagination
+    )
+    page_size_query_param = "page_size"
+
+
 class InstitutionViewSet(viewsets.ModelViewSet):
     queryset = (
         Institution.objects.select_related("jst")
-        .prefetch_related("tags", "parents")
+        .prefetch_related("tags", "parents", "jst", "jst__category")
```

Comment:
Use:

- `'jst__parent', 'jst__parent__parent', 'jst_parent__parent'` to fetch voivodeship efficiency
- move that code to manager, because it is related to `get_jst_voivodeship` defined in `models.py`.

Reply:
As I mentioned earlier, I will use `prefetch_related` with `jst__parent__parent`.
`get_jst_voivodeship` is defined in `InstitutionSerializer`, but I can move it to Institution model so it can be used later.
I can also move this queryset definition to manager (commit in progress).